packages/camera/camera_avfoundation/lib/src/avfoundation_camera.dart

```diff
@@ -291,13 +299,19 @@ class AVFoundationCamera extends CameraPlatform {
   @override
   Stream<CameraImageData> onStreamedFrameAvailable(int cameraId,
       {CameraImageStreamOptions? options}) {
+    _installStreamController(onListen: _onFrameStreamListen);
+    return _frameStreamController!.stream;
+  }
+
+  StreamController<CameraImageData> _installStreamController(
+      {Function()? onListen}) {
     _frameStreamController = StreamController<CameraImageData>(
-      onListen: _onFrameStreamListen,
+      onListen: onListen ?? () {},
       onPause: _onFrameStreamPauseResume,
       onResume: _onFrameStreamPauseResume,
       onCancel: _onFrameStreamCancel,
     );
-    return _frameStreamController!.stream;
+    return _frameStreamController!;
```

Comment:
Why are you returning an ivar here? (ivars should be accessible from the caller already).

Reply:
By returning the variable we remove the need for the caller to _know_ that it is operating on an ivar. This makes refactoring simpler: it could later be changed from being an instance variable to a local variable (or read from another instance), or a subclass could override the implementation, or the return value mocked, and everything will work as before.

Reply:
this would make this function unnecessarily complex, because the side effect of assigning the ivar is not obvious to the caller. I recommend just creating the controller and return it. Then update the ivar in caller site. Something like: 

```
StreamController<CameraImageData> _createStreamController(
      {Function()? onListen})

// caller
_frameStreamController = _createStreamController(...)
```

Reply:
As you wish, I have made that change. Ready to review.